js/models/messages.js

```diff
@@ -2013,9 +2016,18 @@
           } else {
             await conversation.onFriendRequestAccepted();
           }
-          const id = await window.Signal.Data.saveMessage(message.attributes, {
-            Message: Whisper.Message,
-          });
+          let id;
+          // Force save if the message already has an id, used for public channels
+          if (message.id) {
+            id = await window.Signal.Data.saveMessage(message.attributes, {
+              forceSave: true,
+              Message: Whisper.Message,
+            });
+          } else {
+            id = await window.Signal.Data.saveMessage(message.attributes, {
+              Message: Whisper.Message,
+            });
```

Comment:
How about we keep one code path and have
```js
   forceSave: message.id ? true : undefined,
```
(or `false` instead of `undefined`, not sure what's best)

Reply:
Good call, updated